### PR TITLE
Starts game engine

### PR DESCRIPTION
**Pull request by Rolf1e**
Add structures
- [x] Unit
- [x] Skill
- [x] Building
- [x] Player (with wallet)
- [x] Playground
- [x] Some uts and exceptions

### CODE REVIEW

File: rts-core/src/components/playground.rs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+type InnerPlayer = Rc<RefCell<Player>>;
+pub type Players = (InnerPlayer, InnerPlayer);
+
+pub enum PlayerIndex {
+    PlayerOne,
+    PlayerTwo,
+}
+
+pub struct PlayGround {
```
Comment (cafehaine) [inline rts-core/src/components/playground.rs:11]: I guess this means that only two players are supported in a game?
Do you think it'll be easy to switch to an arbitrary number of players in the future?

File: rts-core/src/entity/unit.rs [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    }
+
+    pub fn update_health(&mut self, new_health: i16) -> Result<(), RtsException> {
+        if let Some(res) = Unit::update_attribut_under_max(self.health, self.max_health, new_health)
+        {
+            self.health = res;
+            Ok(())
+        } else {
+            Err(RtsException::GeneralException(String::from(
+                "Failed to update mana",
```
Comment (cafehaine) [inline rts-core/src/entity/unit.rs]: Failed to update health?

**Review by cafehaine — CHANGES_REQUESTED** [review]
2 small nits, looks good to me otherwise!

File: rts-core/src/components/playground.rs (hunk 1 above)
Comment (Rolf1e) [reply rts-core/src/components/playground.rs:11]: I will update with a Vec<Player> you're right !

**Review by cafehaine — APPROVED** [review]
Looks good to me!

**Pull request merged**